scipy/signal/signaltools.py

```diff
@@ -458,16 +512,25 @@ def convolve(in1, in2, mode='full'):
     if volume.ndim == kernel.ndim == 0:
         return volume * kernel
 
+    # see whether the fourier transform convolution method or the direct
+    # convolution method is faster (discussed in scikit-image PR #1792)
+    big_O_constant = 1 / 40.032 if kernel.ndim > 1 else 1 / 1.5
+    direct_time = big_O_constant * np.prod(volume.shape + kernel.shape)
```

Comment:
hmmm...  shouldn't `mode` factor in here?  it doesn't have an effect on fftconvolve but it does on convolve:

```
a = randn(100, 10, 10)

b = randn(100, 10, 10)

timeit fftconvolve(a, b)
100 loops, best of 3: 14.1 ms per loop

timeit fftconvolve(a, b, 'valid')
100 loops, best of 3: 14.6 ms per loop

timeit convolve(a, b)
1 loops, best of 3: 18.7 s per loop

timeit convolve(a, b, 'valid')
1000 loops, best of 3: 326 µs per loop
```